cmd/client_test.go

```diff
@@ -95,59 +101,67 @@ func generateRedisProbeDef(name string) *probes_configpb.ProbeDef {
 
 // TestRegisterAndAddProbe tests that an extension probe type can be registered and added to Cloudprober without error.
 func TestRegisterAndAddProbe(t *testing.T) {
-	testHermes, client := setupCloudproberAndClient()
-	defer teardownCloudproberAndClient(testHermes, client)
+	testHermes, client := setupCloudproberAndClient(t)
+	defer teardownCloudproberAndClient(t, testHermes, client)
 
-	for i := 0; i < len(testProbes); i++ {
-		err := client.RegisterAndAddProbe(int(myprobe.E_RedisProbe.Field), testHermes.Ctx, generateRedisProbeDef(testProbes[i]), &myprobe.Probe{})
-		if err != nil {
-			t.Errorf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
+	for i := 0; i < testRunCount; i++ {
```

Comment:
If you're just looping over teh testProbes, could you use

`for _, probe := range testProbes {`

Reply:
I have removed testProbes as a global variable, and just build the probe name inside the loop using `fmt.Sprintf()`, as the probe name is "testExtension" + <Number>.

This removes a global variable and means that we can increase the number of repetitions by changing one integer variable, rather than adding many new entries to a global array.